tools/camera/camera.go

```diff
@@ -2,109 +2,47 @@ package camera
 
 import (
 	"bytes"
-	"errors"
 	"fmt"
 	"github.com/icholy/digest"
 	"github.com/joschahenningsen/TUM-Live/model"
-	uuid "github.com/satori/go.uuid"
 	"io"
 	"net/http"
 	"os"
-	"strconv"
 	"strings"
 )
 
-//Camera represents AXIS IP cameras the TUM uses
-type Camera struct {
-	Ip   string
-	Auth string
-}
-
-//NewCamera Acts as a constructor for cameras.
-//ip: the ip address of the camera
-//auth: username and password of the camera (e.g. "user:password")
-func NewCamera(ip string, auth string) *Camera {
-	return &Camera{Ip: ip, Auth: auth}
-}
-
-func (c *Camera) TakeSnapshot(outDir string) (filename string, err error) {
-	filename = fmt.Sprintf("%s%s", uuid.NewV4().String(), ".jpg")
-	request, err := c.makeAuthenticatedRequest("GET", "", "/axis-cgi/jpg/image.cgi?compression=75")
-	if err != nil {
-		return "", err
-	}
-	imageFile, err := os.Create(fmt.Sprintf("%s/%s", outDir, filename))
-	if err != nil {
-		return "", err
-	}
-	_, err = imageFile.Write(request.Bytes())
-	if err != nil {
-		return "", err
-	}
-	err = imageFile.Close()
-	if err != nil {
-		return "", err
-	}
-	return filename, nil
-}
-
-//SetPreset tells the camera to use a preset specified by presetId
-func (c Camera) SetPreset(presetId int) error {
-	_, err := c.makeAuthenticatedRequest("GET", "", fmt.Sprintf("/axis-cgi/com/ptz.cgi?gotoserverpresetno=%v&camera=1", presetId))
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
-//GetPresets fetches all presets stored on the camera
-func (c Camera) GetPresets() ([]model.CameraPreset, error) {
-	var presetsForLectureHall []model.CameraPreset
-	resp, err := c.makeAuthenticatedRequest("POST", "action=list&group=root.PTZ.Preset.P0.Position.*.Name", "/axis-cgi/param.cgi")
-	if err != nil {
-		return nil, err
-	}
-	body := resp.String()
-	presets := strings.Split(body, "\n")
-	for _, preset := range presets {
-		if presetSplit := strings.Split(preset, "="); len(presetSplit) == 2 {
-			idParts := strings.Split(presetSplit[0], ".")
-			if len(idParts) != 7 {
-				return nil, errors.New("wrong format for camera preset response")
-			}
-			presetId, err := strconv.ParseInt(strings.Replace(idParts[len(idParts)-2], "P", "", 1), 10, 0)
-			if err != nil {
-				return nil, err
-			}
-			presetsForLectureHall = append(presetsForLectureHall, model.CameraPreset{
-				Name:     presetSplit[1],
-				PresetID: int(presetId),
-			})
-		}
-	}
-	return presetsForLectureHall, nil
+type Cam interface {
```

Comment:
Since I am currently working on Preset utility mocks, could you please add `//go:generate mockgen -source=camera.go -destination ../mock_tools/camera.go` at the top? See #526 for more information. Thanks 🚀